### ISSUE TITLE
ibmqx5 initial_layout for multiple registers

### ISSUE THREAD

**Issue by ogamel** [QISKit/qiskit-sdk-py#522]
Hello. 
I am now running on the ibmqx5. However, when I have two quantum registers, I get an error: 
```
line 473, in swap_mapper "DAGCircuit" % (k[0], k[1]))
qiskit.mapper._mappererror.MapperError: 'initial_layout qubit qr[1] not in input DAGCircuit'
```

I have traced it up the chain, and believe the issue is in the compile function in _compiler.py. num_qubits is calculated to be the total number of qubits in all registers. But initial_layout is defined such that each register has num_qubits qubits, which leads to errors. (edited)

E.g. if both registers (called 'qr' and 'qar') have just one qubit each, the current code calculates

`initial_layout =  {('qar', 0): ('q', 0),  ('qar', 1): ('q', 1),  ('qr', 0): ('q', 0),  ('qr', 1): ('q', 1)}
`leading to errors, since qar[1] and qr[1] don't exist.

The error is resolved when I define by hand
`initial_layout={('qar', 0): ('q', 1), ('qr', 0): ('q', 0)}`

**Comment by nonhermitian** [comment]
Are you using the latest 0.5.3 release?

**Comment by ogamel** [comment]
I am currently away from my machine, but I believe it is 0.5.3. At the
oldest it may be 0.5.2.

**Comment by nonhermitian** [comment]
There was a recent fix for that issue i. 0.5.3.

**Comment by ajavadia** [comment]
@ogamel can you please close the issue if the problem was resolved in 0.5.3?

**Comment by ogamel** [comment]
Yes going from 0.5.2 to 0.5.3 fixed it. 
Thank you
